generic3g/actions/AccumulatorAction.F90

```diff
@@ -188,4 +206,49 @@ subroutine accumulate_R4(this, update_field, rc)
 
    end subroutine accumulate_R4
 
+   !============================= HOOK METHODS =================================
+   ! These are hook methods that can be overwritten by extending types.
+
+   subroutine update_pre(this, rc)
+      class(AccumulatorAction), intent(inout) :: this
+      integer, optional, intent(out) :: rc
+      
+      integer :: status
+
+      _RETURN(_SUCCESS)
+
+   end subroutine update_pre
+
+   subroutine clear_post(this, rc)
+      class(AccumulatorAction), intent(inout) :: this
+      integer, optional, intent(out) :: rc
+      
+      integer :: status
+
+      _RETURN(_SUCCESS)
+      _UNUSED_DUMMY(this)
+      
+   end subroutine clear_post
+   
```

Comment:
Starting to be a bit of a code smell that there are so many no-op methods on the base class.   Not for now,  but I may eventually (ha) explore use of has-a vs is-a (current) implementation for the individual cases.

Reply:
Yes, that's pretty much what I'm hinting at.  But stay the course for now.  Functionality before perfection.